### ISSUE TITLE
Suggestion Multiply by a function result

### ISSUE THREAD

**Issue by Koslas1** [geel9/runelite-custom-item-hovers#5]
As a suggestion to build on from #4
I currently have this as one of my item hovers
"items": ["Blessed bone shards"],
      "hovers": [
        [
		  "  Xp gained",
          "  \u2022  <%qtymult(5)%> xp (<%qtydivide(100)%>x Sunfire wine)",
          "  \u2022  <%qtymult(6)%> xp (<%qtydivide(400)%>x Blessed wine) "
        ]
      ]
    }

I currently have 1,788 blessed bone shards, which for the 3rd line results in 
10,728 xp (4x Blessed wine)
I was experimenting with the functions, and was unable to achieve this. and was wondering if there was a way to make it divide by 400, then multiply that by 400, therefore resulting in a "1,600" to know that I could only use 1,600 of my 1,788 blessed bone shards? It would save me using my brain or a calculator to be able to do <%qtydivide(400)%> * 400 as I know how much wine I am able to use already. It would also allow me to work out how much actual xp I can gain from that amount of wine
So I could do something like "<%qtymult(6)%> xp (<%mult(%qtydivide(400)%> * 400) * 6%> xp)" which would show "10,728 xp (9,600 xp)" to show I currently have 10,728 banked xp, but only 9,600 I could currently get

Also I like to thank you for this plugin as I use it quite a bit, and it's an amazing plugin

**Comment by geel9** [comment]
I'm actually working on a big update that will allow for much more flexible mathematical expressions, a la:

`${qty * 5} xp (${floor(qty / 100)}x Sunfire Wine`

Supports any arbitrarily-complex expression with parentheses, functions, and variables.


Alongside this I'm also adding support for "tables", which are files where you can define variables on items by their item name. Then, any hover on that item has access to those variables in their hover text.

For example if you have a lot of really similar hovers that only differ based on some numerical values, you could create a table defining that xp variable on each item, then just create a single hover that applies to all the items, and reference the xp variable.

Also toying around with UI support in the plug in but that's a ways away tbh

**Comment by Koslas1** [comment]
So I'd be able to easily set one up with all my bones, and then just change the value of how much blessed bone shards and xp they would give total, as each bone gives a different amount?
Also would you be able to make it so you could do if functions for stuff like if magicLevel < 90 then 
"Cannot reanimate this head"
Else
"Xp gained: " etc? Would make it handy for sharing around custom hovers and also can be setup so when you get a certain Magic level it then changes the item hover

**Comment by geel9** [comment]
I'll show an example.

You set up a table file (this one is `customitemhovers/tables/hunter_meats.json`):

```json
{
	"name": "hunter_meats",
	"is_table_file": "absolutely",
	"items": {
		"Cooked wild kebbit": {
			"$also": "Raw wild kebbit",
			"immediate_heal": "4",
			"delayed_heal": "4",
			"quetzal_charges": "1",
			"cooking_xp": "73"
		},
		"Cooked larupia": {
			"$also": "Raw larupia",
			"immediate_heal": "6",
			"delayed_heal": "5",
			"quetzal_charges": "1",
			"cooking_xp": "92"
		},
		"Cooked barb-tailed kebbit": {
			"$also": "Raw barb-tailed kebbit",
			"immediate_heal": "7",
			"delayed_heal": "5",
			"quetzal_charges": "1",
			"cooking_xp": "106"
		},
		"Cooked graahk": {
			"$also": "Raw graahk",
			"immediate_heal": "8",
			"delayed_heal": "8",
			"quetzal_charges": "2",
			"cooking_xp": "124"
		},
		"Cooked kyatt": {
			"$also": "Raw kyatt",
			"immediate_heal": "9",
			"delayed_heal": "8",
			"quetzal_charges": "2",
			"cooking_xp": "143"
		},
		"Cooked pyre fox": {
			"$also": "Raw pyre fox",
			"immediate_heal": "11",
			"delayed_heal": "8",
			"quetzal_charges": "2",
			"cooking_xp": "154"
		},
		"Cooked sunlight antelope": {
			"$also": "Raw sunlight antelope",
			"immediate_heal": "12",
			"delayed_heal": "9",
			"quetzal_charges": "3",
			"cooking_xp": "175"
		},
		"Cooked dashing kebbit": {
			"$also": "Raw dashing kebbit",
			"immediate_heal": "13",
			"delayed_heal": "10",
			"quetzal_charges": "3",
			"cooking_xp": "215"
		},
		"Cooked moonlight antelope": {
			"$also": "Raw moonlight antelope",
			"immediate_heal": "14",
			"delayed_heal": "12",
			"quetzal_charges": "3",
			"cooking_xp": "220"
		}
	}
}
```

Then you just set up hovers for all the items you want to target:

```json
{
  "is_hover_map": "absolutely",
  "hovers": [
     {
      "items_regex": ["Raw ((wild|barb-tailed|dashing) kebbit|larupia|graahk|kyatt|pyre fox|(sunlight|moonlight) antelope)"],
      "hovers": [
        [
            "<col=E447FF>Cooking:</col> ${cooking_xp * qty} xp (${cooking_xp} ea)",
            "Quetzal Whistle Charges: ${quetzal_charges}"
        ]
      ]
    },
    {
      "items_regex": ["(Raw|Cooked) ((wild|barb-tailed|dashing) kebbit|larupia|graahk|kyatt|pyre fox|(sunlight|moonlight) antelope)"],
      "hovers": [
        [
          "<col=36FF0D>Heals:</col> ${immediate_heal + delayed_heal} (${immediate_heal} + ${delayed_heal})"
        ]
      ]
    }
  ]
}
```


Alongside any variables you specify, you also have access to the same built-in variables from before: `qty`, `id`, `price`, and `high_alch`.

**Comment by geel9** [comment]
Regarding conditional statements: it's something I've been thinking about for sure, but doing it elegantly and in a performant way is going to be tricky.

**Comment by geel9** [comment]
Resolved by #6 -- will need to be submitted to the hub and approved before you can use it, though
